CHEMH-245: Halfpill title banner

jenbreese opened the pull request:
# READY FOR REVIEW
- Ready now for real review: @jdwjdwjdw would you take a look and let me know what you see? 

# Summary
- A new version of the halfpill to be used in the Hero Banner location
- Verify the regular halfpill banner looks better in the various responsive views.

# Review By (Date)
- 9/26

# Criticality
- Normal

# Urgency
- Normal

# Review Tasks

## Setup tasks and/or behavior to test

1. Check out this branch
2. Rebuild Cache and import config `drush cr ; drush ci`
3. Navigate to a page and add a hero banner with a Halfpill Title variant
4. 
<img width="1543" alt="Screenshot 2023-09-19 at 9 33 45 AM" src="https://github.com/SU-SWS/chem_h_subtheme/assets/1187335/c83b9ea9-59e4-4ec8-8905-5395c0e18ad5">

5. Add a page title (what you want for the h1), description, and button
6. save.
7. Open it in layout builder. Remove the title section. Do this for just this page and not the whole template.
8. 
<img width="1634" alt="Screenshot 2023-09-19 at 9 34 44 AM" src="https://github.com/SU-SWS/chem_h_subtheme/assets/1187335/50d70bab-b8e2-4082-8498-aa2123263745">

9. Verify it looks like this. 
10. Verify it has an h1 and no h2 in the banner.
11. 
<img width="1637" alt="Screenshot 2023-09-19 at 9 33 59 AM" src="https://github.com/SU-SWS/chem_h_subtheme/assets/1187335/35d797fa-9c7d-466d-9900-70c67af4bc8c">


### Site Configuration Sync

- Is there a config:export in this PR that changes the config sync directory?

## Front End Validation
- [ ] Design is approved by @ user?
- [ ] HTML validation: Is the markup using the appropriate semantic tags and [passes validation](https://validator.w3.org/nu/)? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Cross-browser testing: Has been performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Automated accessibility: Scans performed? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?
- [ ] Manual accessibility: Manually tested? Or, [QA request ticket created](https://github.com/SU-SWS/template_warehouse/blob/master/jira_templates/QA_request_template.txt)?

## Backend / Functional Validation
### Code
- [ ] Are the naming conventions following our standards?
- [ ] Does the code have sufficient inline comments?
- [ ] Is there anything in this code that would be hidden or hard to discover through the UI?
- [ ] Are there any [code smells](https://blog.codinghorror.com/code-smells/)?
- [ ] Are tests provided? eg (unit, behat, or codeception)

### Code security
- [ ] Are all [forms properly sanitized](https://www.drupal.org/docs/8/security/drupal-8-sanitizing-output)?
- [ ] Any obvious [security flaws or new areas for attack](https://www.drupal.org/docs/8/security)?

## General
- [ ] Is there anything included in this PR that is not related to the problem it is trying to solve?
- [ ] Is the approach to the problem appropriate?

# Affected Projects or Products
- Does this PR impact any particular projects, products, or modules?

# Associated Issues and/or People
[- CHEMH-245](https://stanfordits.atlassian.net/browse/CHEMH-245)

# Resources
- [AMP Tool](https://stanford.levelaccess.net/index.php)
- [Accessibility Manual Test Script](https://docs.google.com/document/d/1ZXJ9RIUNXsS674ow9j3qJ2g1OAkCjmqMXl0Gs8XHEPQ/edit?usp=sharing)
- [HTML Validator](https://validator.w3.org/)
- [Browserstack](https://live.browserstack.com/dashboard) and link to [Browserstack Credentials](https://asconfluence.stanford.edu/confluence/display/SWS/External+Account+Credentials)

jdwjdwjdw reviewed (commented):
Everything's looking great to me! Only suggestion I have - I think it'd be nice to add some additional space between the image and the heading badge on mobile, similar to how the other halfpill variant looks.
![Screenshot 2023-09-19 at 12 26 48 PM](https://github.com/SU-SWS/chem_h_subtheme/assets/49299083/50d4112d-0d91-4e3c-990e-a6526d70d544)
![Screenshot 2023-09-19 at 12 27 10 PM](https://github.com/SU-SWS/chem_h_subtheme/assets/49299083/43a2059e-14f7-49d0-98d8-58b81fb0bd8c)

jenbreese commented:
@jdwjdwjdw thanks! I fixed your comment, then tried the text right and left. I added the 20px rounded edge when 2500px + wide. I think it is ready for real PR review now.

jdwjdwjdw reviewed (approved):
Looking good to me! 🚀
